pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/jstl/JstlTypeConverter.java

```diff
@@ -88,8 +90,14 @@ protected Long coerceToLong(Object value) {
     if (value instanceof Time) {
       return ((Time) value).toLocalTime().toNanoOfDay() / 1_000_000;
     }
+    if (value instanceof Timestamp) {
+      return ((Timestamp) value).getTime();
+    }
     if (value instanceof Date) {
-      return ((Date) value).getTime();
+      return ((Date) value).getTime() / MILLIS_PER_DAY;
```

Comment:
I think j.u.Date should stay as a timestamp. We only do the change for LocalDate